drivers/comparator/comparator_shell.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/comparator.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/kernel.h>
+
+#include <stdlib.h>
+
+#define AWAIT_TRIGGER_DEFAULT_TIMEOUT \
+	CONFIG_COMPARATOR_SHELL_AWAIT_TRIGGER_DEFAULT_TIMEOUT
+
+#define AWAIT_TRIGGER_MAX_TIMEOUT \
+	CONFIG_COMPARATOR_SHELL_AWAIT_TRIGGER_MAX_TIMEOUT
+
+/* Mapped 1-1 to enum comparator_trigger */
+static const char *const trigger_lookup[] = {
+	"NONE",
+	"RISING_EDGE",
+	"FALLING_EDGE",
+	"BOTH_EDGES",
+};
+
+static K_SEM_DEFINE(triggered_sem, 0, 1);
+
+static int get_device_from_str(const struct shell *sh,
```

Comment:
temping to suggest that this and the timeout one should go in `subsys/shell/shell_utils.c`, just an idea, maybe a followup

Reply:
agreed, but as a followup :) I'm sure these additions can be added tons of places in the tree, we should do that in one PR :)

Reply:
yup, cool